runtime/common/src/impls.rs

```diff
@@ -23,16 +20,43 @@ use sp_std::vec::Vec;
 
 common_types::impl_tranche_token!();
 
-pub struct DealWithFees<Config>(PhantomData<Config>);
-pub type NegativeImbalance<Config> =
-	<Balances<Config> as Currency<<Config as SystemConfig>::AccountId>>::NegativeImbalance;
-impl<Config> OnUnbalanced<NegativeImbalance<Config>> for DealWithFees<Config>
+pub type NegativeImbalance<R> = <pallet_balances::Pallet<R> as Currency<
+	<R as frame_system::Config>::AccountId,
+>>::NegativeImbalance;
+
+struct ToAuthor<R>(sp_std::marker::PhantomData<R>);
+impl<R> OnUnbalanced<NegativeImbalance<R>> for ToAuthor<R>
 where
-	Config: AuthorshipConfig + BalancesConfig + SystemConfig,
+	R: pallet_balances::Config + pallet_authorship::Config,
 {
-	fn on_nonzero_unbalanced(amount: NegativeImbalance<Config>) {
-		if let Some(who) = Authorship::<Config>::author() {
-			Balances::<Config>::resolve_creating(&who, amount);
+	fn on_nonzero_unbalanced(amount: NegativeImbalance<R>) {
+		if let Some(author) = <pallet_authorship::Pallet<R>>::author() {
+			<pallet_balances::Pallet<R>>::resolve_creating(&author, amount);
+		}
+	}
+}
+
+pub struct DealWithFees<R>(sp_std::marker::PhantomData<R>);
+impl<R> OnUnbalanced<NegativeImbalance<R>> for DealWithFees<R>
+where
+	R: pallet_balances::Config + pallet_treasury::Config + pallet_authorship::Config,
+	pallet_treasury::Pallet<R>: OnUnbalanced<NegativeImbalance<R>>,
+{
+	fn on_unbalanceds<B>(mut fees_then_tips: impl Iterator<Item = NegativeImbalance<R>>) {
+		if let Some(fees) = fees_then_tips.next() {
+			// for fees, split the destination
+			let mut split = fees.ration(
+				super::TREASURY_FEE_RATIO.deconstruct(),
+				(Perbill::one() - super::TREASURY_FEE_RATIO).deconstruct(),
+			);
+			if let Some(tips) = fees_then_tips.next() {
+				// for tips, if any, 100% to author
+				tips.merge_into(&mut split.1);
+			}
+
+			use pallet_treasury::Pallet as Treasury;
+			<Treasury<R> as OnUnbalanced<_>>::on_unbalanced(split.0);
+			<ToAuthor<R> as OnUnbalanced<_>>::on_unbalanced(split.1);
```

Comment:
Would it be an idea to name the `split` 0 and 1 parts to make this more readable? Like 

```
let (treasury_amt, author_amt) =  ...

// and then here
```suggestion
			<Treasury<R> as OnUnbalanced<_>>::on_unbalanced(treasury_amt);
			<ToAuthor<R> as OnUnbalanced<_>>::on_unbalanced(author_amt);
```

Reply:
I left as Polkadot did, but I agree with that, good idea! 😃